custom-vu/src/main/kotlin/jces1209/vu/page/DcAddScreenShot.kt

```diff
@@ -0,0 +1,32 @@
+package jces1209.vu.page
+
+import jces1209.vu.wait
+import org.openqa.selenium.By
+import org.openqa.selenium.Keys
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.interactions.Actions
+import org.openqa.selenium.support.ui.ExpectedConditions
+
+class DcAddScreenShot(
+    driver: WebDriver
+) : AttachScreenShot(driver) {
+    override var screenShotLocator = By.xpath("(//*[@class='attachment-thumb])")
+    override fun pasteScreenShot(): Int {
+        countBefore = getIssueScreenShotsCount(screenShotLocator)
+        Actions(driver).keyDown(Keys.CONTROL).sendKeys("v").perform()
+        driver.wait(
+            ExpectedConditions.elementToBeClickable(
+                By.className("aui-button aui-button-primary")))
+            .click()
+        driver.wait(
+            ExpectedConditions.invisibilityOfElementLocated(By.id("attach-screenshot-dialog")))
+        driver.wait(ExpectedConditions.numberOfElementsToBeMoreThan(
+            screenShotLocator, countBefore))
+        driver.wait(ExpectedConditions.visibilityOfAllElementsLocatedBy(
```

Comment:
It's redundant because it's being checked in the previous wait

Reply:
is it possible that element can be in DOM but still invisible?

Reply:
good point. let's leave it

Reply:
[In general, yes](https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/support/ui/ExpectedConditions.html#visibilityOf-org.openqa.selenium.WebElement-):
> Visibility means that the element is not only displayed but also has a height and width that is greater than 0.

In the case of this specific element: I don't know.